ObservatoryExplorer/DefaultCriteria.cs

```diff
@@ -61,13 +49,23 @@ public static List<(string Description, string Detail, bool SystemWide)> CheckIn
             #region Value Checks
             if (settings.HighValueMappable)
             {
-                if (HighValueTerraformablePlanetClasses.Contains(scan.PlanetClass) && scan.TerraformState?.Length > 0)
-                {
-                    results.Add("High-Value Mapping", $"{scan.DistanceFromArrivalLS:0}Ls, {scan.PlanetClass} (TF)");
-                }
-                if (HighValueNonTerraformablePlanetClasses.Contains(scan.PlanetClass) && scan.TerraformState?.Length == 0)
+                IList<string> HighValueNonTerraformablePlanetClasses = new string[] {
```

Comment:
I'm curious why not leave this as static (other than keeping to keep it next to the criteria)? I'd really not want to waste time allocating and garbage collecting it again every time we process a scan. (Those aren't free/cheap actions.) Because this is a mutable object instance, const won't help either.

Reply:
I am unconvinced that these are expensive actions. I think the .NET compiler is smarter about this than you give it credit for.

```c#
using System;
using System.Collections.Generic;
using System.Diagnostics;

namespace ListAllocationTest
{
    internal class Program
    {
        static void Main(string[] args)
        {
            StaticList staticList = new();
            InlineList inlineList = new();
            
            for (int i = 0; i < 5; i++)
            {
                long staticCount = staticList.Iterate();
                long inlineCount = inlineList.Iterate();
                Console.WriteLine($"Static iterations: {staticCount}");
                Console.WriteLine($"Inline iterations: {inlineCount}");
                Console.WriteLine($"Static speed relative to inline: {(double)staticCount / inlineCount:N4}x");
            }
        }
    }

    internal class StaticList
    {
        private static IList<string> HighValueNonTerraformablePlanetClasses = new string[] {
            "Earthlike body",
            "Ammonia world",
            "Water world",
        };

        public long Iterate()
        {
            Stopwatch stopwatch = new();
            stopwatch.Start();
            long iterations = 0;
            List<string> list = new();
            while (stopwatch.ElapsedMilliseconds < 10000)
            {
                list.Add(HighValueNonTerraformablePlanetClasses[(int)(stopwatch.ElapsedMilliseconds % 3)]);
                iterations++;
            }
            stopwatch.Reset();
            return iterations;
        }
    }

    internal class InlineList
    {
        public long Iterate()
        {
            Stopwatch stopwatch = new();
            stopwatch.Start();
            long iterations = 0;
            List<string> list = new();
            while (stopwatch.ElapsedMilliseconds < 10000)
            {
                IList<string> HighValueNonTerraformablePlanetClasses = new string[] {
                    "Earthlike body",
                    "Ammonia world",
                    "Water world",
                };
                list.Add(HighValueNonTerraformablePlanetClasses[(int)(stopwatch.ElapsedMilliseconds % 3)]);
                iterations++;
            }
            stopwatch.Reset();
            return iterations;
        }
    }
}
```

Output:
```
Static iterations: 5836702
Inline iterations: 5871449
Static speed relative to inline: 0.9941x
Static iterations: 5916428
Inline iterations: 5863030
Static speed relative to inline: 1.0091x
Static iterations: 5914297
Inline iterations: 5887645
Static speed relative to inline: 1.0045x
Static iterations: 5906464
Inline iterations: 5877156
Static speed relative to inline: 1.0050x
Static iterations: 5904397
Inline iterations: 5876337
Static speed relative to inline: 1.0048x
```

For the sake of a non-reliable half-percent of performance in a worst-case scenario where the other work being done per iteration is trivial in comparison to what Explorer does, I think I'd rather keep it nearby in the code for organizational purposes.

Reply:
Far more thorough response than I expected, so I tip my hat to you. :)

I'll do my best to give C# the benefit of the doubt in the future. :D